test/metabase/search/filter_test.clj

```diff
@@ -83,49 +88,53 @@
-           "past3hours~"                             [[:>= :card.created_at #t "2023-05-04T07:00"]
-                                                      [:< :card.created_at #t "2023-05-04T11:00"]]
-           "next3days"                               [[:>= [:cast :card.created_at :date] #t "2023-05-05"]
-                                                      [:< [:cast :card.created_at :date]  #t "2023-05-08"]]
-           "thisminute"                              [[:>= :card.created_at #t "2023-05-04T10:02"]
-                                                      [:< :card.created_at #t "2023-05-04T10:03"]]
-           "lasthour"                                [[:>= :card.created_at #t "2023-05-04T09:00"]
-                                                      [:< :card.created_at #t "2023-05-04T10:00"]]
-           "past1months-from-36months"               [[:>= [:cast :card.created_at :date] #t "2020-04-01"]
-                                                      [:< [:cast :card.created_at :date]  #t "2020-05-01"]]
-           "today"                                   [[:>= [:cast :card.created_at :date] #t "2023-05-04"]
-                                                      [:< [:cast :card.created_at :date] #t "2023-05-05"]]
-           "yesterday"                               [[:>= [:cast :card.created_at :date] #t "2023-05-03"]
-                                                      [:< [:cast :card.created_at :date] #t "2023-05-04"]]))))
+    ;; testing with different clock to ensure created at filter always built at 'UTC'
+    ;; regardless of what the system timezone is
+    (doseq [clock [#t "2023-05-04T10:02:05Z[UTC]" #t "2023-05-04T19:02:05Z[Asia/Tokyo]"]]
+      (mt/with-clock clock
+        (test.util.misc/with-fix-local-date-time-at-zone! (str (.getZone ^ZonedDateTime clock))
+          (are [created-at expected-where]
+               (= expected-where
+                  (->> (search.filter/build-filters
+                        base-search-query "card"
+                        (merge default-search-ctx {:created-at created-at}))
+                       :where
+                       ;; drop the first 2 clauses [:and [:= :card.archived false]]
+                       (drop 2)))
+               ;; absolute datetime
+               "Q1-2023"                                 [[:>= :card.created_at #t "2023-01-01T00:00Z[UTC]"]
+                                                          [:< :card.created_at #t "2023-04-01T00:00Z[UTC]"]]
+               "2016-04-18~2016-04-23"                   [[:>= :card.created_at #t "2016-04-18T00:00Z[UTC]"]
+                                                          [:< :card.created_at #t "2016-04-24T00:00Z[UTC]"]]
+               "2016-04-18"                              [[:>= :card.created_at #t "2016-04-18T00:00Z[UTC]"]
+                                                          [:< :card.created_at #t "2016-04-19T00:00Z[UTC]"]]
```

Comment:
I'm realising now I'm confused by the purpose of this change. 

The goal should be that the user should be able to search for something like “all questions created in 2023-02-01 in the Asia/Tokyo timezone”, where Asia/Tokyo is the browser time zone? That should get compiled to SQL with a comparison with timestamps with time zone UTC like this. But that's not what's being tested here, since there is no browser time zone input into the test, only the date string.

Is this an intermediate step towards being able to handle browser time zones?